### PR TITLE
Change wording for dev/deploy

### LINKED ISSUE #6387: reword help and documentation messages to not use "deploy" in context with `odo dev`

### PR DESCRIPTION
**Pull request by feloy**
**What type of PR is this:**

/kind bug

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6387

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

**Comment by feloy** [comment]
These changes will modify lots of commands outputs displayed in doc. 
Blocked by #6192

File: cmd/odo/help_test.go [hunk 1]
```diff
@@ -29,8 +29,8 @@ Examples:
 
 	mainCommands = `Main Commands:
   build-images Build images
-  deploy       Deploy components
-  dev          Deploy component to development cluster
+  deploy       Run components in Deploy mode
```
Comment (rm3l) [inline cmd/odo/help_test.go:32]: To be consistent with the help message for `dev`:

```suggestion
  deploy       Run your application on the cluster in the Deploy mode
```

**Review by rm3l — CHANGES_REQUESTED** [review]

**Review by rm3l — APPROVED** [review]

**Comment by feloy** [comment]
/override OpenShift-Integration-tests/OpenShift-Integration-tests
Flaky e2e test

**Pull request merged**